app/lib/service/topics/models.dart

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:json_annotation/json_annotation.dart';
+import 'package:logging/logging.dart';
+import 'package:meta/meta.dart';
+import 'package:path/path.dart' as p;
+import 'package:pub_dev/frontend/static_files.dart';
+import 'package:yaml/yaml.dart';
+
+part 'models.g.dart';
+
+final _log = Logger('topics');
+
+late final canonicalTopics = () {
+  try {
+    final f = File(p.join(resolveAppDir(), '../doc/topics.yaml'));
+    final content = CanonicalTopicFileContent.fromYaml(f.readAsStringSync());
+    content.verifyContent();
+    return content;
+  } on Exception catch (e, st) {
+    _log.shout('failed to load doc/topics.yaml', e, st);
+
+    // This is sad, but we can just ignore it!
+    return CanonicalTopicFileContent(topics: <CanonicalTopic>[]);
+  }
+}();
+
+@JsonSerializable(
+  checked: true,
+  explicitToJson: true,
+  disallowUnrecognizedKeys: true,
+)
+class CanonicalTopicFileContent {
+  final List<CanonicalTopic> topics;
+
+  CanonicalTopicFileContent({
+    required this.topics,
+  });
```

Comment:
Why not?
```suggestion
  CanonicalTopicFileContent({
    required this.topics,
  }) {
    verifyContent();
  }
```
possibly even just copy pasting the checks into this constructor, so we don't need a `verifyContent` method.

Reply:
Done.